common/darwin/Classes/AudioUtils.m

```diff
@@ -23,7 +23,7 @@ + (void)ensureAudioSessionWithRecording:(BOOL)recording {
                        error:nil];
         [session setMode:config.mode error:nil];
         [session unlockForConfiguration];
-    } else if (!recording || (session.category == AVAudioSessionCategoryAmbient
+    } else if (!recording && (session.category == AVAudioSessionCategoryAmbient
```

Comment:
@cloudwebrtc do you recall why this was changed from `&&` to `||` in https://github.com/flutter-webrtc/flutter-webrtc/pull/1162? There was [a comment](https://github.com/flutter-webrtc/flutter-webrtc/pull/1162/files#r1020637849) indicating that a custom audio session configuration wouldn't be overwritten since `config.categoryOptions = 0` was removed; however, that line was reintroduced in https://github.com/flutter-webrtc/flutter-webrtc/commit/954ff2a702c13e15cc6fc7edcaa1c9215e27b3dc.